pkg/apis/core/validation/validation_test.go

```diff
@@ -16948,6 +16964,81 @@ func fmtErrs(errs field.ErrorList) string {
 	return buf.String()
 }
 
+func verifyVersionedValidationEquivalence(t *testing.T, obj, old k8sruntime.Object) {
+	t.Helper()
+
+	// Accumulate errors from all versioned validation, per version.
+	all := map[string]field.ErrorList{}
+	accumulate := func(t *testing.T, gv string, errs field.ErrorList) {
+		all[gv] = errs
+	}
+	if old == nil {
+		runtimetest.RunValidationForEachVersion(t, legacyscheme.Scheme, sets.Set[string]{}, obj, accumulate)
+	} else {
+		runtimetest.RunUpdateValidationForEachVersion(t, legacyscheme.Scheme, sets.Set[string]{}, obj, old, accumulate)
+	}
```

Comment:
I've favored having a separate validation and update-validation functions for consistency with how strategy is defined, but it has led to an increase of functions in the scheme, and in these test utilities.  Should we reconsider this?  The alternative would be to switch to a conversion where a nil oldObject always indicates create validation, and a non-nil oldObject indicates update validation.

Reply:
If it cuts down on duplication, I think the semantic is plenty obvious.  But I am very close to it...